store/tikv/2pc.go

```diff
@@ -717,3 +734,18 @@ func appendBatchBySize(b []batchKeys, region RegionVerID, keys [][]byte, sizeFn
 	}
 	return b
 }
+
+// markBatchAsPrimary marks the batch that contains the primaryKey as primary and return it
+func markBatchAsPrimary(batches []batchKeys, primaryKey []byte) (primary batchKeys, others []batchKeys) {
+	for i := range batches {
+		for _, k := range batches[i].keys {
```

Comment:
If the primary is found, for the rest of the batches, you don't need to check their keys any more.

Reply:
Got your idea, thanks, let me fix it.

Reply:
@disksing Sorry for responding so late!

I got stuck on a tiny bug that would miss a batch when do prewrite, and then there are too many things took away my attention from the PR.

Now the bug has been fixed, take another review please, thanks!